test/uk/gov/hmrc/tradergoodsprofilesrouter/service/RouterServiceSpec.scala

```diff
@@ -0,0 +1,441 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.service
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import uk.gov.hmrc.http.HeaderCarrier
+import play.api.http.Status._
+import uk.gov.hmrc.tradergoodsprofilesrouter.connectors.EISConnector
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.eis.GetEisRecordsResponse
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.errors.{Error, ErrorResponse, RouterError}
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ApplicationConstants
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class RouterServiceSpec extends AnyWordSpec with Matchers {
+
+  val eisConnector = new EISConnector {
```

Comment:
should not have code that is not implemented. Also here we are testing the wrong thing. We should test the fechRecord function as this is the public interface. The other are and should be all private function and they will be tested indirectly throw fechRecord. We can do this together if you want.